nodebuilder/pruner/module.go

```diff
@@ -34,31 +33,42 @@ func ConstructModule(tp node.Type, cfg *Config) fx.Option {
 		// This is necessary to invoke the pruner service as independent thanks to a
 		// quirk in FX.
 		fx.Invoke(func(_ *pruner.Service) {}),
-		fx.Provide(func(store *eds.Store) pruner.Pruner {
-			return full.NewPruner(store)
-		}),
 	)
 
 	switch tp {
 	// TODO: Eventually, light nodes will be capable of pruning samples
 	//  in which case, this can be enabled.
 	case node.Light:
+		if cfg.EnableService {
+			return fx.Module("prune",
+				baseComponents,
+				prunerService,
+				fx.Provide(light.NewPruner),
+			)
+		}
 		return fx.Module("prune",
 			baseComponents,
 			fx.Provide(light.NewPruner),
```

Comment:
why provide light pruner if serv not enabled?

Reply:
It is not gree, so it used to be this way 😆  Will remove

Reply:
The comment got moved or something. The comment is about if case above